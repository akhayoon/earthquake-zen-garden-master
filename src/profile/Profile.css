.avatar {
  width: 12rem;
}

.column {
  display: flex;
  flex-direction: column;
  flex: 0
}

.two-columns {
  display: flex;
  flex-direction: column;
  flex-basis: 100%;
  flex: 2
}
