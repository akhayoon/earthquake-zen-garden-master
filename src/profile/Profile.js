import React from 'react';
import earthQuakeData from '../data/earthquakes.json';
import Field from '../common/components/Fields';
import './Profile.css';

export default function Profile() {
  const { profile } = earthQuakeData;

  return (
    <div className='container'>
      <h1>Profile</h1>
      <div className='row'>
        <div className='column'>
          <img className="avatar" src={profile.avatarImage} />
        </div>
        <div className='two-columns'>
            <div className='row'>
              <Field label='First Name' value={profile.firstName}/>
              <Field label='Last Name' value={profile.lastName}/>
              <Field label='Phone' value={profile.phone}/>
              <Field label='Email' value={profile.email}/>
              <Field label='Bio' value={profile.bio}/>
          </div>
        </div>
      </div>
    </div>
  );
}
