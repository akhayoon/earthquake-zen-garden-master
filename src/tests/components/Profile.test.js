import React from 'react';
import { describe, expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import Profile from '../../profile/Profile';

describe('Test Profile Component', () => {
  test('it should match Snapshot', () => {
    const tree = renderer
      .create(<Profile />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
