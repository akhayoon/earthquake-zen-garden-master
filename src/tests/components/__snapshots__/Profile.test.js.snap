// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile Component it should match Snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Profile
  </h1>
  <div
    className="row"
  >
    <div
      className="column"
    >
      <img
        className="avatar"
        src="https://upload.wikimedia.org/wikipedia/commons/5/59/That_Poppy_profile_picture.jpg"
      />
    </div>
    <div
      className="two-columns"
    >
      <div
        className="row"
      >
        <div
          className="row"
        >
          <div
            className="field title"
          >
            First Name
          </div>
          <div
            className="field value"
          >
            Sally
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="field title"
          >
            Last Name
          </div>
          <div
            className="field value"
          >
            Wang
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="field title"
          >
            Phone
          </div>
          <div
            className="field value"
          >
            01-343-989-2345
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="field title"
          >
            Email
          </div>
          <div
            className="field value"
          >
            sw@nowhere.ic.kp
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="field title"
          >
            Bio
          </div>
          <div
            className="field value"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
